2.1/slidingWindow.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"math/rand"
+	"sort"
+)
+
+type PercentileCalculator interface {
+	RecordValue(uint)
+	GetPercentile() uint
+}
+
+type R struct {
+	data       seedData
+	percentile float32
+	window     uint
+}
+
+type seedData []uint
```

Comment:
nit: seedData seems like a misnomer.  There isn't anything "seed"y about this, it is just the current internal state for the PercentileCalculator.